statsd/statsd.go

```diff
@@ -552,6 +563,17 @@ func (c *Client) Gauge(name string, value float64, tags []string, rate float64)
 	return c.send(metric{metricType: gauge, name: name, fvalue: value, tags: tags, rate: rate, globalTags: c.tags, namespace: c.namespace})
 }
 
+// Gauge measures the value of a metric at a given time.
+// Even with client side aggregation enabled, there is no aggregation done on a metric
+// sent using GaugeWithTimestamp: it is written as is in the serialization buffer.
+func (c *Client) GaugeWithTimestamp(name string, value float64, tags []string, rate float64, timestamp time.Time) error {
+	if c == nil {
+		return ErrNoClient
+	}
+	atomic.AddUint64(&c.telemetry.totalMetricsGauge, 1)
```

Comment:
Do we want a dedicated telemetry for metrics without timestamp ? Do we want to know what % of traffic is with timestamp ?

Reply:
For now I don't think we have a reason adding a new metrics (sent by the client to the intake by every clients...) just for this.
I think this is an open question that I have to solve: I open a ticket for this but I think the best thing to do is to use the same telemetry for now (it's not wrong neither to count it in the "total metrics gauge" and "total metrics count").

Reply:
Sounds good